aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -852,11 +852,17 @@ private boolean modelExists(Model model, QueryPredicate predicate) throws DataSt
      * @return the Model instance from SQLite, if it exists, otherwise null.
      */
     private Model query(Model model) {
+        final String modelName = getModelName(model);
+        final ModelSchema schema = modelSchemaRegistry.getModelSchemaForModelClass(modelName);
+        final SQLiteTable table = SQLiteTable.fromSchema(schema);
+        final String primaryKeyName = table.getPrimaryKeyColumnName();
+        final QueryPredicate matchId = QueryField.field(primaryKeyName).eq(model.getId());
```

Comment:
Seems like we always have to build out some scaffolding to parse the model. Maybe we need a helper data structure. Could be like `SqlStorageData data = SqlStorageData.from(model)`? Not for this PR, of course.

Reply:
I agree! Building a whole `SQLiteTable` instance from model schema every time a model is being queried or saved is such an unnecessary amount of work. It was meant to be a temporary solution while in a rush to release it a year ago, but it definitely doesn't cut it now.